core/modules/xrdsvc/SsiProvider.h

```diff
@@ -41,28 +53,29 @@ class SsiProviderServer : public XrdSsiProvider
 {
 public:
 
-    virtual XrdSsiService *GetService(XrdSsiErrInfo& eInfo, char const* contact, int oHold=256) {
+    virtual XrdSsiService *GetService(XrdSsiErrInfo& eInfo,
+                                      char const* contact, int oHold=256) {
         return _service.get();
     }
 
     virtual bool Init(XrdSsiLogger* logP, XrdSsiCluster* clsP, char const* cfgFn,
-            char const* parms, int argc, char **argv) {
-        _service = std::unique_ptr<SsiService>(new SsiService(logP));
-        return true;
-    }
+                      char const* parms, int argc, char** argv);
 
-    virtual rStat QueryResource(char const*rName, char const* contact=0) {
-        // Not called on this object but part of pure virtual interface,
-        // so we need to provide at least this dummy implementation.
-        return isPresent;
-    }
+    virtual rStat QueryResource(char const* rName, char const* contact=0);
+
+                   SsiProviderServer() : _cmsSsi(0), _logSsi(0) {}
+    virtual      ~SsiProviderServer() {}
```

Comment:
If you want destructor inline then I believe that for all `unique_ptr` members the template types must be complete, otherwise it should fail to compile (if you include this header into a file which has no definition for ChunkInventory). It's better to move destructoor definition to .cc file.